xarray/core/datatree.py

```diff
@@ -636,31 +629,31 @@ def __array__(self, dtype=None):
             "invoking the `to_array()` method."
         )
 
-    def __repr__(self) -> str:
-        return formatting.datatree_repr(self)
+    def __repr__(self) -> str:  # type: ignore[override]
```

Comment:
This is another annotation I wanted to call out. I think it makes sense here. `NamedNode.__repr__` has an optional `level` kwarg, which isn't an argument in `repr_datatree`. `repr_datatree` is using `RenderTree`, which _does_ have a `maxLevel`, but I don't think that's quite the same thing. (If it is, though, we could edit `DataTree.__repr__` to match the signature of `NamedNode.__repr__`, and then pass it on through)

I don't think properly maps to the `maxLevel`